openwisp_monitoring/utils.py

```diff
@@ -22,10 +22,9 @@ def wrapper(*args, **kwargs):
         for attempt_no in range(1, max_retries + 1):
             try:
                 return method(*args, **kwargs)
-            except Exception as err:
-                logger.info(
-                    f'Error while executing method "{method.__name__}":\n{err}\n'
-                    f'Attempt {attempt_no} out of {max_retries}.\n'
+            except Exception as e:
+                logger.exception('Error while executing method "%s": %s. Attempt %n out of %n.',
+                    method.__name__, e, attempt_no, max_retries
```

Comment:
can you please keep the f-string usage which is more readable?

Reply:
By using a template string though we get the string value even if we will not log the message. This way the computation of the message is delayed until the logging system decides that it should actually do the work of formatting the string.

I'm not sure how much runtime inspection the logger does but I guess it will also work (print a summary stack-trace) without having the current exception as an argument.

Reply:
@nemesifier  I changed it back to using template strings as logger.exception does the runtime stack checking and can figure out the exception stacktrace to print.